src/doc/reference.md

```diff
@@ -2114,10 +2114,19 @@ Sometimes one wants to have different compiler outputs from the same code,
 depending on build target, such as targeted operating system, or to enable
 release builds.
 
-There are two kinds of configuration options, one that is either defined or not
-(`#[cfg(foo)]`), and the other that contains a string that can be checked
-against (`#[cfg(bar = "baz")]`). Currently, only compiler-defined configuration
-options can have the latter form.
+There are two kinds of configuration options:
+
+- Boolean options, which are either on or off, e.g. `#[cfg(foo)]`. These can
+  either be defined by the compiler or specified on the command line, e.g.
+  `rustc --cfg foo`.
+- Key-value options, where a key may be associated with one or more values, e.g.
+  `#[cfg(bar = "baz")]`. These can also either be defined by the compiler or
+  specified on the command line, e.g. `rustc --cfg 'bar="baz"'` (whitespace around
+  the `=` is ignored).
+
+one that is either defined or not
+(`#[cfg(foo)]`), and the other that takes on one or more string values
+(`#[cfg(bar = "baz")]`).
```

Comment:
better but why should we say there are two kinds of configuration options, when there is just one? `first` and `second="x"` are both just names of `cfg` vars. Also some old text left in here at the end.